functional_tests/features/bundle_ui.feature

```diff
@@ -0,0 +1,133 @@
+        Given there is a <role> user
+        When the <role> logs in
+        Then the user can create a bundle
+        And the user can see the created bundle
+
+        Examples: bundles
+           | role               |
+#           | Publishing Admin   |
+#           | Publishing Officer |
+#
+#
+    Scenario Outline: A User cannot create a bundle
+        Given there is a <role> user
+        When the <role> logs in
+        Then the user cannot create a bundle
+        Examples: bundles
+           | role     |
+#           | Viewer   |
+
+#---- Bundle Edit UI Tests -----
+
+    Scenario Outline: A User can edit a bundle
+        Given there is a <Role> user
+        And there is a <Creator Role> user
+        And there are 1 Statistical Analysis pages
+        And there are 1 release calendar pages
+        And there is a preview team
+        And the <Role> is a member of the preview team
+        And there are <number_of_bundles> bundles with <Bundle_Details>
+        When the <Role> logs in
+        Then the user can find the bundle
+        And the user can edit the bundle
+
+        Examples: bundles
+           | number_of_bundles | Role                       | Creator Role        | Bundle_Details                                                                                                                                            |
+#           | 1                 | Publishing Admin           | Publishing Admin    | {"Role": "Publishing Admin",   "Creator Role": "Publishing Admin",   "status": "Draft", "preview_teams": false, "add_rel_cal": false, "add_stat_page": false} |
+#           | 1                 | Publishing Admin           | Publishing Officer  | {"Role": "Publishing Admin",   "Creator Role": "Publishing Officer", "status": "Draft", "preview_teams": false, "add_rel_cal": false, "add_stat_page": false} |
+#           | 1                 | Publishing Officer         | Publishing Admin    | {"Role": "Publishing Officer", "Creator Role": "Publishing Admin",   "status": "Draft", "preview_teams": false, "add_rel_cal": false, "add_stat_page": false} |
+#           | 1                 | Publishing Officer         | Publishing Officer  | {"Role": "Publishing Officer", "Creator Role": "Publishing Officer", "status": "Draft", "preview_teams": false, "add_rel_cal": false, "add_stat_page": false} |
+
+
+#---- Bundle Preview UI Tests -----
+
+  Scenario Outline: A User can preview a bundle
+        Given there is a <Role> user
+        And there is a <Creator Role> user
+        And there are 1 Statistical Analysis pages
+        And there are 1 release calendar pages
+        And there is a preview team
+        And the <Role> is a member of the preview team
+        And there are <number_of_bundles> bundles with <Bundle_Details>
+        When the <Role> logs in
+        Then the user can see the Bundles ready for preview
+        Then the <Role> can preview a bundle
+
+      Examples: bundles
```

Comment:
am confused of what's going on here? And if this is needed?

As I been debugging line by line the behave steps, and using the playwight inspector to check for assertions etc, and so far, this is not making sense. Assertions seems to be confusing and some assertions are incomplete as it's checking for half of a message or text on a page etc.